### ISSUE TITLE
Zuul does not include ACAO header in response to OPTIONS request even when the proxied service has @CrossOrigin set

### ISSUE THREAD

**Issue by abhishekbiyala** [spring-cloud/spring-cloud-netflix#923]

**Comment by dsyer** [comment]
I don't think the Spring MVC stack handles OPTIONS without some configuration, so you are probably just running into a filter or a condition in Spring MVC somewhere before you even reach Zuul. Does it work if you go through the Zuul servlet (default path /zuul)?

**Comment by abhishekbiyala** [comment]
I ended up creating a CorsFilter as a workaround:

```
	@Override
	public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
			throws IOException, ServletException {

		HttpServletRequest req = (HttpServletRequest) request;
		HttpServletResponse res = (HttpServletResponse) response;

		if("OPTIONS".equalsIgnoreCase(req.getMethod())) {
		    res.setHeader("Access-Control-Allow-Origin", "*");
		    res.setHeader("Access-Control-Allow-Methods", "POST, PUT, GET, OPTIONS, DELETE, PATCH");
		    res.setHeader("Access-Control-Allow-Headers", "origin, content-type, accept, x-requested-with");
		    res.setHeader("Access-Control-Max-Age", "3600");
		} else {
			chain.doFilter(request, response);
		}
	}
```

However, would like to see first class support for this in Spring Cloud Zuul for routing OPTIONS request instead of creating custom Filter.
The source code with steps to reproduce the original issue is located at: https://github.com/abhishekbiyala/zuul-http-options-issue

**Comment by abhishekbiyala** [comment]
@dsyer It works if I go through Zuul servlet (default path /zuul). It would help if this feature is documented well, especially if there are any side effects of using Zuul servlet directly.

**Comment by spencergibb** [comment]
@abhishekbiyala how would you expect zuul to support this when you use zuul through the spring mvc dispatcher?

**Comment by abhishekbiyala** [comment]
@spencergibb not sure if this is feasible, but ideally I would like zuul (even when using spring mvc dispatcher) to let OPTIONS request pass thru by setting a flag under zuul.routes properties set.

**Comment by spencergibb** [comment]
I don't think we should be doing anything beyond what spring mvc does here.

**Comment by dsyer** [comment]
I'm not sure I agree. A proxy should be able to relay OPTIONS and let the backend decide how to handle it.

**Comment by dsyer** [comment]
Fixed in 7f075d1. But see also #1423.
